src/internet/radiobrowser/radiobrowserservice.cpp

```diff
@@ -0,0 +1,448 @@
+    parentItem->appendRow(item);
+  }
+}
+
+void RadioBrowserService::PopulateStreams(QStandardItem* parentItem,
+                                          StreamList& streams) {
+  if (parentItem->hasChildren())
+    parentItem->removeRows(0, parentItem->rowCount());
+
+  for (const Stream& stream : streams) {
+    QStandardItem* item = new QStandardItem(
+        IconLoader::Load("icon_radio", IconLoader::Lastfm), QString());
+    item->setText(stream.name_);
+    item->setData(QVariant::fromValue(stream.ToSong(name_)),
+                  InternetModel::Role_SongMetadata);
+    item->setData(InternetModel::PlayBehaviour_SingleItem,
+                  InternetModel::Role_PlayBehaviour);
+    item->setData(InternetModel::Type_Track, InternetModel::Role_Type);
+
+    parentItem->appendRow(item);
+  }
+}
+
+void RadioBrowserService::ReloadSettings() {
+  QSettings s;
+  s.beginGroup(kSettingsGroup);
+
+  main_server_url_ =
+      s.value("server", RadioBrowserService::defaultServer).toString();
+
+  if (root_ && root_->hasChildren()) {
+    root_->removeRows(0, root_->rowCount());
+    root_->setData(true, InternetModel::Role_CanLazyLoad);
+  }
+}
+
+bool RadioBrowserService::EnsureServerConfig() {
+  if (main_server_url_.isEmpty())
+    return false;
+  else
+    return true;
+}
+
+void RadioBrowserService::LastRequestFailed() { ShowConfig(); }
+
+void RadioBrowserService::ShowConfig() {
+  app_->OpenSettingsDialogAtPage(SettingsDialog::Page_RadioBrowser);
+}
+
+void RadioBrowserService::Search(int search_id, const QString& query,
+                                 const int limit) {
+  QString determinedUrl =
+      RadioBrowserService::SearchUrl.arg(main_server_url_, query).arg(limit);
+  QUrl url(determinedUrl);
+
+  QNetworkReply* reply = network_->get(QNetworkRequest(url));
+  int task_id = app_->task_manager()->StartTask(tr("Getting channels"));
+
+  connect(reply, &QNetworkReply::finished,
+          [=] { this->SearchFinishedInternal(reply, task_id, search_id); });
+}
+
+void RadioBrowserService::SearchFinishedInternal(QNetworkReply* reply,
+                                                 int task_id, int search_id) {
+  app_->task_manager()->SetTaskFinished(task_id);
+  reply->deleteLater();
+  QJsonDocument document = ParseJsonReply(reply);
+
+  StreamList list;
+  QJsonArray contents = document.array();
+  qLog(Debug) << "RadioBrowser station list found:" << contents.size();
+  for (const QJsonValue& c : contents) {
+    QJsonObject item = c.toObject();
+    ReadStation(item, &list);
+  }
+
+  emit SearchFinished(search_id, list);
+}
+
+void RadioBrowserService::ResolveStationUrl(const QUrl& original_url) {
+  QString determinedUrl = RadioBrowserService::PlayClickUrl.arg(
+      main_server_url_, original_url.path());
+  QUrl url(determinedUrl);
+
+  QNetworkReply* reply = network_->get(QNetworkRequest(url));
+  int task_id = app_->task_manager()->StartTask(tr("Getting station"));
+
+  connect(reply, &QNetworkReply::finished, [=] {
```

Comment:
If you're using lambdas, you don't really need the member functions. You can do all the work in the lambda.

Reply:
Done, thank you